STM32F405RGTx/Core/Src/rtos_threads/svpwm_thread.cpp

```diff
@@ -0,0 +1,100 @@
+#include <stdio.h>
+#include <stdint.h>
+#include "main.h"
+#include "threads.hpp"
+#include "math.h"
+
+SVPWM::SVPWM(float SwitchFreq, VHZPROFILE *vhz, TIM_HandleTypeDef *htim){
+	ModIndex = 0;
+	Freq = 0;
+	SwitchFreq = SwitchFreq;
+	Alpha = 0;
+	NewEntry = 0;
+	SectorPointer = 0;
+	U = 0;
+	V = 0;
+	W = 0;
+	StepAngle = 0;
+	EntryOld = 0;
+	ta = 0;
+	tb = 0;
+	to = 0;
+
+	vhz = vhz;
+	htim = htim;
+}

```

Comment:
Can you move these to the static initialize function? We shouldn't realistically be instantiating an instance of this object ever since it's all static. You should get rid of the constructor all together